File: csharp/IRepository.cs
using System.Collections.Generic;

namespace TestDouble
{
public interface IRepository<K,T> {
	 T getById(K id);
	 List<T> getAll();
	 bool add(T element);
	 T pop(K id);
	 bool removeById(K id);
	 bool remove(T element);
	 bool contains(K id);
	 T getNew();
}
}

File: java/src/IRepository.java
import java.util.List;

public interface IRepository<K,T> {
	public T getById(K id);
	public List<T> getAll();
	public boolean add(T element);
	public T pop(K id);
	public boolean removeById(K id);
	public boolean remove(T element);
	public boolean contains(K id);
	public T getNew();
}
